File: src/services/audit.ts
import { commands, type AuditEvent } from "@/bindings";
import { unwrap } from "./tauri";

export const auditService = {
  list: (limit = 200) =>
    unwrap(commands.listAuditEvents(limit)) as Promise<AuditEvent[]>,
  export: () => unwrap(commands.exportAuditLog()) as Promise<string>,
  clear: async () => {
    await unwrap(commands.clearAuditLog());
  },
};

File: src-tauri/src/commands/audit.rs
use std::sync::Arc;

use tauri::State;

use crate::audit::AuditEvent;
use crate::error::AppResult;
use crate::state::AppState;

#[tauri::command]
#[specta::specta]
pub fn list_audit_events(
    limit: usize,
    state: State<'_, Arc<AppState>>,
) -> AppResult<Vec<AuditEvent>> {
    state.audit.read_recent(limit)
}

#[tauri::command]
#[specta::specta]
pub fn export_audit_log(state: State<'_, Arc<AppState>>) -> AppResult<String> {
    state.audit.export_text()
}

#[tauri::command]
#[specta::specta]
pub fn clear_audit_log(state: State<'_, Arc<AppState>>) -> AppResult<()> {
    state.audit.clear()
}
